stateproof/builder.go

```diff
@@ -357,29 +357,8 @@ func (spw *Worker) broadcastSigs(brnd basics.Round, proto config.ConsensusParams
 	}
 }
 
-func lowestRoundToRemove(currentHdr bookkeeping.BlockHeader) basics.Round {
-	proto := config.Consensus[currentHdr.CurrentProtocol]
-	nextStateProofRnd := currentHdr.StateProofTracking[protocol.StateProofBasic].StateProofNextRound
-	if proto.StateProofInterval == 0 {
-		return nextStateProofRnd
-	}
-
-	recentRoundOnRecoveryPeriod := basics.Round(uint64(currentHdr.Round) - uint64(currentHdr.Round)%proto.StateProofInterval)
-	oldestRoundOnRecoveryPeriod := recentRoundOnRecoveryPeriod.SubSaturate(basics.Round(proto.StateProofInterval * proto.StateProofMaxRecoveryIntervals))
-	// we add +1 to this number since we want exactly StateProofMaxRecoveryIntervals elements in the history
-	oldestRoundOnRecoveryPeriod++
-
-	var oldestRoundToRemove basics.Round
-	if oldestRoundOnRecoveryPeriod > nextStateProofRnd {
-		oldestRoundToRemove = oldestRoundOnRecoveryPeriod
-	} else {
-		oldestRoundToRemove = nextStateProofRnd
-	}
-	return oldestRoundToRemove
-}
-
-func (spw *Worker) deleteOldSigs(currentHdr bookkeeping.BlockHeader) {
-	oldestRoundToRemove := lowestRoundToRemove(currentHdr)
+func (spw *Worker) deleteOldSigs(currentHdr *bookkeeping.BlockHeader) {
+	oldestRoundToRemove := GetOldestExpectedStateProof(currentHdr)
```

Comment:
do I understand it right: if `GetOldestExpectedStateProof` returns `oldestRoundToRemove=0` then we can delete unconditionally? This is right in a general case but what about the following edge cases:
1. StateProof is disabled with a protocol upgrade but there are still older intervals to confirm?
2. If StateProof not enabled, is it possible the table does not exist and can deletion statement fails?

Reply:
the node always creates the DB file and the table. so I don't think it is a problem.
if `GetOldestExpectedStateProof`  returns 0 than actually nothing gets removed.
no signature  or builder would have rnd <0

or am I missing something